### ISSUE TITLE
Add support for publishing artifacts to a separate branch

### ISSUE THREAD

**Issue by nick-woodward** [semantic-release/semantic-release#857]
### TL;DR;

Prior to version 13 it was possible to publish to a separate branch via [publish-latest](https://www.npmjs.com/package/publish-latest).
Version 13 now validates the current version via Git instead of NPM which leads to 
issues when releasing from `master` and also publishing to the `latest` branch on GitHub.

### I've got time...

One of the libraries I've developed ([salte-auth](https://github.com/salte-io/salte-auth)), supports being installed via npm and bower. 
My primary motivator for this was maintaining support for Polymer 1 and 2 which both require packages to be installed via bower.

As we all know, bower requires that packages are published to a git repository (preferably with a tag).
It's always been my personal preference not to taint my master branch with build artifacts. So instead I opted to publish to a `latest` branch. This workflow was facilitated by using `semantic-release` in combination with a package developed by Kent Dodds (I'm avoiding tagging him since it isn't overly relevant) called [publish-latest](https://www.npmjs.com/package/publish-latest). 

Up to version 12 of `semantic-release` his package worked flawlessly. However, version 13 introduced some amazing features into `semantic-release`. One of those was opting to make the default version validation plugin use Git instead of NPM. Unfortunately, in my case, this leads to validation issues since all of my versions are published on the latest branch and my deployments are made from the master branch.

## New feature description

It'd be awesome if `semantic-release` could release from one branch and push to another on Git.

**Comment by pvdlg** [comment]
If I understand correctly you want to have a `latest` branch with a commit that contains your built assets and to tag this commit with the version number?

If it's the case I don't think it's even possible. semantic-release will tag the last commit of the a release on `master` and use it later on to figure out the last release.
Even if it wasn't a semantic-release requirement it's still a recommended practice to tag your commits in `master` with the version. Among other things it allow you to checkout a particular version tag and get the code as it was on that release. For example you can do `git checkout tags/v1.0.0` to test if `v1.0.0` has a particular bug.
That also allow you to create hotfixes on old versions. For example if your last version is `2.0.0` and a user experience a bug on version `1.1.0` and cannot upgrade to `2.0.0` for certain reasons, then you can do `git checkout tags/v1.1.0 -b 1.1.x` and fix the bug on the branch `1.1.x` and make a `1.1.1` release from there.

I imagine your previous workflow was tagging on `latest` but not on `master`?

The problem is that tags name have to be unique in Git and point to one commit. So you cannot have a `v1.0.0` tag that point to the last commit included in the release on `master` and another `v1.0.0` tag that point to a release tag on `latest`.

So I don't think we can do anything to support that workflow in semantic-release for several reasons:
- We do not want to support a workflow that allow to not tag on `master` for reason explains above
- You situation is specific to `bower` which is not supported anymore
- An alternative solution is to push your assets to the `master` branch, which seems to be an acceptable compromise as long as you have to publish on `bower`

**Comment by nick-woodward** [comment]
@pvdlg 
Thanks for taking the time to provide some well articulated feedback!

I'm actually bringing my source code with me when I checkout the `latest` branch. So the source code exists on both `latest` and `master`, but only `latest` has the built artifact. So technically hot fixes should be possible, but could create some weird situations given the branch split.

Migrating my code from `latest` back to `master` was my first option (and likely the one I'll move forward with), however I figured it was worth a shot to see how something like this would be received over here. I totally don't blame you for not wanting to take on any debt related to `bower`.

Again, thanks for your time!

**Comment by pvdlg** [comment]
If you find a solution that works you can still implement it with custom plugins or with `@semantic-release/exec`. The goal of having plugins has always been to allow implementing specific workflow that are not common enough to be part of the plugins we maintain.
